sys/kern/src/arch/riscv32.rs

```diff
@@ -569,8 +560,22 @@ fn trap_handler(task: &mut task::Task) {
         Trap::Exception(Exception::InstructionFault) => unsafe {
             handle_fault(task, FaultInfo::IllegalText);
         },
+
         _ => {
-            panic!("Unimplemented exception");
+            if mcause.is_exception() == true {
+                panic!("Unimplemented exception");
+            } else if (mcause.cause()
+                == Trap::Interrupt(Interrupt::MachineExternal))
+                || (mcause.code() >= 16)
```

Comment:
Is the `platform_interrupt_handler` supposed to catch all interrupts besides the timer?  So if the interrupt is not caught by this `if` then a task cannot  use it? 
If this is the case, I think the `16` should be a `12`, since anything `>=12` is a reserved interrupt.

Reply:
Along those same lines, is there a compile time check to ensure a task does not attempt to register an unsupported interrupt number?

Reply:
The `platform_interrupt_handler` is supposed to handle the interrupts designated for platform use. Interrupts 12-15 are reserved by the ISA, so I think for now it's best to leave it as is. 

We don't have a compile-time check at the moment, but that's something that can be added, I'll put it in the kernel's `build.rs`.

Reply:
Where do you see interrupts >15 are for platform, while 12-15 are for ISA?
Under the mcause section, it has any interrupt >=12 is reserved.